### PR TITLE
feat: add array util to remove duplicates

### PR DESCRIPTION
**Pull request by CatchMe2**
## Changes

Please describe

## Checklist

- [x] Apply one of following labels; `major`, `minor`, `patch` or `skip-release`
- [x] I've updated the documentation, or no changes were necessary
- [x] I've updated the tests, or no changes were necessary

### CODE REVIEW

File: src/utils/arrayUtils.ts [hunk 1]
```diff
@@ -42,3 +42,12 @@ export function removeFalsy<const T>(
 ): T[] {
   return array.filter((e) => e) as T[]
 }
+
+/**
+ * Return a copy of the given array without duplicates.
+ */
+export function removeDuplicates<const T>(
```
Comment (kibertoad) [inline src/utils/arrayUtils.ts:49]: can you add documentation to readme?

File: src/utils/arrayUtils.spec.ts [hunk 2]
```diff
@@ -64,4 +64,12 @@ describe('arrayUtils', () => {
       expect(result).toEqual(['valid', 1])
     })
   })
+
+  describe('removeDuplicates', () => {
+    it('should remove all duplicates', () => {
```
Comment (kibertoad) [inline src/utils/arrayUtils.spec.ts:69]: would be nice to check how objects are handled as well

Comment (kibertoad) [reply src/utils/arrayUtils.spec.ts:69]: also null and undefined

Comment (CatchMe2) [reply src/utils/arrayUtils.spec.ts:69]: Good idea, added

File: src/utils/arrayUtils.ts (hunk 1 above)
Comment (CatchMe2) [reply src/utils/arrayUtils.ts:49]: Good catch, added!

File: docs/array-utils.md [hunk 3]
```diff
@@ -32,3 +32,13 @@ Return a copy of the given array without falsy values (eg: false, 0, '', null, u
 const array = ['', false, null, 'valid', 1, undefined, 0]
 console.log(removeFalsy(array)) // result: ['valid', 1]
 ```
+
+`removeDuplicates<const T>(array: readonly T[]): T[]`
+
+Return a copy of the given array without duplicates.
```
Comment (kibertoad) [inline docs/array-utils.md:38]: might be helpful to mention that object deduplication works by reference, not deep equality

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
thank you!

**Pull request merged**